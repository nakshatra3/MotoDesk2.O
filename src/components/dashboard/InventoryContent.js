import React, { useState, useEffect } from "react"; // Import useEffect
import {
  Box,
  Tabs,
  Tab,
  TextField,
  Button,
  Select,
  MenuItem,
  InputLabel,
  FormControl,
  Typography,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  Paper,
  TableContainer,
} from "@mui/material";
import { FaShoppingCart, FaTruckLoading } from "react-icons/fa";

const inventoryData = {
  totalAvailableCars: 120,
  totalOrderedCars: 30,
};

function InventoryContent() {
  const [activeTab, setActiveTab] = useState(0); // 0 = Order Car, 1 = Order Tracking
  const [formData, setFormData] = useState({
    name: "",
    model: "",
    color: "",
    quantity: 0,
  });
  const [responseMessage, setResponseMessage] = useState("");
  const [ordersData, setOrdersData] = useState([]); // State for orders data

  const [inventoryData, setInventoryData] = useState([]); // State for inventory data


  useEffect(() => {
    // Fetch orders from backend when component mounts
    const fetchOrders = async () => {
      try {
        const response = await fetch("https://motodesk2-o.onrender.com/order/getOrder");
        const data = await response.json();
        setOrdersData(data.orders); // Set orders data
      } catch (error) {
        console.error("Error fetching orders:", error);
      }
    };
    const fetchInventory = async () => {
      try {
        const response = await fetch("https://motodesk2-o.onrender.com/inventory/"); // Adjust the endpoint as needed
        const data = await response.json();
        setInventoryData(data); // Set inventory data
      } catch (error) {
        console.error("Error fetching inventory:", error);
      }
    };

    fetchOrders();
    fetchInventory(); // Call the fetch function
  }, []);

  const handleTabChange = (event, newValue) => {
    setActiveTab(newValue);
  };

  const getStatusColor = (status) => {
    switch (status) {
      case "confirmed":
        return "green";
      case "pending":
        return "orange";
      case "delivered":
        return "blue";
      default:
        return "black";
    }
  };

  const handleInputChange = (e) => {
    const { name, value } = e.target;
    setFormData({ ...formData, [name]: value });
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const response = await fetch("https://motodesk2-o.onrender.com/order/add", {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify(formData),
      });

      if (response.ok) {
        const result = await response.text();
        setResponseMessage(result);
      } else {
        const errorData = await response.json();
        setResponseMessage(errorData.message || "An error occurred");
      }
    } catch (error) {
      console.error("Error:", error);
      setResponseMessage("Server Error");
    }
  };

  const renderOrderForm = () => {
    return (
      <Box
        component="form"
        sx={{ display: "flex", flexDirection: "column", gap: 2 }}
        onSubmit={handleSubmit}
      >
        <TextField
          label="Name"
          name="name"
          value={formData.name}
          onChange={handleInputChange}
          required
          fullWidth
        />
        <FormControl fullWidth>
          <InputLabel>Model</InputLabel>
          <Select
            name="model"
            value={formData.model}
            onChange={handleInputChange}
            required
          >
            <MenuItem value="top">top</MenuItem>
            <MenuItem value="mid">mid</MenuItem>
            <MenuItem value="base">base</MenuItem>
          </Select>
        </FormControl>
        <TextField
          label="Colour"
          name="color"
          value={formData.color}
          onChange={handleInputChange}
          required
          fullWidth
        />
        <TextField
          label="Quantity"
          name="quantity"
          type="number"
          value={formData.quantity}
          onChange={handleInputChange}
          required
          fullWidth
        />
        <Box sx={{ display: "flex", justifyContent: "center", gap: 2 }}>
          <Button
            variant="contained"
            color="primary"
            sx={{ maxWidth: "200px" }}
            type="submit"
          >
            Place Order
          </Button>
        </Box>
        {responseMessage && (
          <Typography
            color="primary"
            variant="h6"
            sx={{ textAlign: "center", mt: 2 }}
          >
            {responseMessage}
          </Typography>
        )}
      </Box>
    );
  };

 
  const renderOrderTrackingTable = () => (
    <TableContainer component={Paper} sx={{ maxHeight: 400 }}>
      <Table stickyHeader aria-label="order tracking table">
        <TableHead>
          <TableRow>
            <TableCell>Order ID</TableCell>
            <TableCell>Name</TableCell>
            <TableCell>Model</TableCell>
            <TableCell>Cost Price</TableCell>
            <TableCell>Color</TableCell>
            <TableCell>Quantity</TableCell>
            <TableCell>Status</TableCell>
            <TableCell>Total Cost</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {ordersData.map((order) => (
            <TableRow key={order._id}>
              <TableCell>{order._id}</TableCell>
              <TableCell>{order.name}</TableCell>
              <TableCell>{order.model}</TableCell>
              <TableCell>₹{order.costPrice || "N/A"}</TableCell>
              <TableCell>{order.color}</TableCell>
              <TableCell>{order.quantity}</TableCell>
              <TableCell>
                <Typography sx={{ color: getStatusColor(order.status), fontWeight: "bold" }}>
                  {order.status}
                </Typography>
              </TableCell>
              <TableCell>
                ₹{(order.costPrice || 0) * order.quantity}
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
  

  return (
    <Box
      sx={{
        display: "flex",
        gap: 4,
        p: 4,
        height: "100vh",
        backgroundColor: "#eef2f6",
      }}
    >
      <Box sx={{ flex: 2, maxWidth: "100%", minWidth: "700px", width:"700px"}}>
        <Tabs
          value={activeTab}
          onChange={handleTabChange}
          variant="fullWidth"
          centered
          sx={{
            "& .MuiTab-root": {
              color: "#3f51b5",
              "&:hover": { color: "white", backgroundColor: "#3f51b5" },
            },
          }}
        >
          <Tab icon={<FaShoppingCart />} label="Order Car" />
          <Tab icon={<FaTruckLoading />} label="Order Tracking" />
        </Tabs>
        <Box
          sx={{
            mt: 2,
            p: 3,
            boxShadow: 3,
            borderRadius: 2,
            backgroundColor: "white",
            overflowY: "auto",
            maxHeight: "80vh",
          }}
        >
          {activeTab === 0 ? renderOrderForm() : renderOrderTrackingTable()}
        </Box>
      </Box>

      <Box sx={{ flex: 1, display: "flex", flexDirection: "column", gap: 2 }}>
        <Box
          sx={{
            display: "flex",
            alignItems: "center",
            gap: 2,
            p: 2,
            boxShadow: 2,
            borderRadius: 2,
            backgroundColor: "white",
          }}
        >
          <FaShoppingCart style={{ fontSize: "40px", color: "#4caf50" }} />
          <Box>
            <Typography variant="h6">Total Available Cars</Typography>
            <Typography variant="h4" color="primary">
            {inventoryData.reduce((total, inv) => total + inv.quantity, 0)}
            </Typography>
          </Box>
        </Box>
        <Box
          sx={{
            display: "flex",
            alignItems: "center",
            gap: 2,
            p: 2,
            boxShadow: 2,
            borderRadius: 2,
            backgroundColor: "white",
          }}
        >
          <FaShoppingCart style={{ fontSize: "40px", color: "#ff9800" }} />
          <Box>
            <Typography variant="h6">Total Ordered Cars</Typography>
            <Typography variant="h4" color="primary">
            {ordersData.reduce((total, order) => total + order.quantity, 0)}
            </Typography>
          </Box>
        </Box>
      </Box>
    </Box>
  );
}

export default InventoryContent;
