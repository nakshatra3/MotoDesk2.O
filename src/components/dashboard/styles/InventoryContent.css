/* Main Layout */
.inventory-layout {
  display: flex;
  gap: 20px;
  padding: 20px;
  height: 100vh;
  background-color: #eef2f6;
  box-sizing: border-box;
}

/* Left Section: MUI Tabs and Form */
.left-section {
  flex: 2;
  display: flex;
  flex-direction: column;
  gap: 20px;
  width: calc(100% - 400px);
  max-width: 60%;
  min-width: 500px;
}

/* Form Container */
.form-container {
  padding: 30px;
  border-radius: 10px;
  box-shadow: 0 3px 12px rgba(0, 0, 0, 0.1);
  overflow-y: auto;
  max-height: 80vh;
  width: auto;
}

/* Inventory Summary Cards */
.right-section {
  flex: 1;
  display: flex;
  flex-direction: column;
  gap: 20px;
}

.summary-card {
  display: flex;
  align-items: center;
  gap: 20px;
  background-color: white;
  padding: 20px;
  border-radius: 10px;
  box-shadow: 0 4px 12px rgba(0, 0, 0, 0.1);
  transition: transform 0.2s;
}

.summary-card:hover {
  transform: scale(1.05);
}

.summary-icon {
  font-size: 40px;
}

.available .summary-icon {
  color: #4caf50;
}

.ordered .summary-icon {
  color: #ff9800;
}

.summary-card h4 {
  margin: 0;
  font-weight: bold;
}

.summary-card p {
  margin: 0;
  font-size: 24px;
  color: #1a237e;
}

/* Form Content */
.form-content {
  display: flex;
  flex-direction: column;
  gap: 20px;
}

.form-group input,
.form-group select {
  width: 100%;
  max-width: 600px;
  padding: 12px;
  border: 1px solid #ccc;
  border-radius: 8px;
}

.action-button {
  background-color: #1976d2;
  color: white;
  padding: 14px;
  border: none;
  border-radius: 8px;
  cursor: pointer;
  font-size: 16px;
  width: 100%;
  max-width: 300px;
  align-self: center;
  transition: background-color 0.3s;
}

.action-button:hover {
  background-color: #1565c0;
}
