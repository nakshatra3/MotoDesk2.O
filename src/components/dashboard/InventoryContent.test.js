import React from 'react';
import { render, screen } from '@testing-library/react';
import InventoryContent from './InventoryContent'; // Adjust path as necessary
import '@testing-library/jest-dom';

describe('InventoryContent Component', () => {
  test('renders InventoryContent component', async () => {
    // Mocking fetch calls or other setup code goes here
    global.fetch = jest.fn(() =>
      Promise.resolve({
        json: () => Promise.resolve({ orders: [], inventory: [] }), // Adjust based on your API
      })
    );

    const { asFragment } = render(<InventoryContent />);
    
    // You can use await screen.findByText to wait for async components to load
    const availableCarsHeader = await screen.findByText(/Total Available Cars/i);
    expect(availableCarsHeader).toBeInTheDocument();

    // Take a snapshot
    expect(asFragment()).toMatchSnapshot();
  });
});
