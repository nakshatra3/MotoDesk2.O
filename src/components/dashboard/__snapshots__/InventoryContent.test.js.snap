// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryContent Component renders InventoryContent component 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-gbxtch"
  >
    <div
      class="MuiBox-root css-13qhm9n"
    >
      <div
        class="MuiTabs-root css-193jq17-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-s2t35c-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            class="MuiTabs-flexContainer MuiTabs-centered css-stp9o6-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-labelIcon MuiTab-textColorPrimary MuiTab-fullWidth Mui-selected css-nmkn4i-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="0"
              type="button"
            >
              <svg
                class="MuiTab-iconWrapper MuiTab-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
                />
              </svg>
              Order Car
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-labelIcon MuiTab-textColorPrimary MuiTab-fullWidth css-nmkn4i-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <svg
                class="MuiTab-iconWrapper MuiTab-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 640 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M50.2 375.6c2.3 8.5 11.1 13.6 19.6 11.3l216.4-58c8.5-2.3 13.6-11.1 11.3-19.6l-49.7-185.5c-2.3-8.5-11.1-13.6-19.6-11.3L151 133.3l24.8 92.7-61.8 16.5-24.8-92.7-77.3 20.7C3.4 172.8-1.7 181.6.6 190.1l49.6 185.5zM384 0c-17.7 0-32 14.3-32 32v323.6L5.9 450c-4.3 1.2-6.8 5.6-5.6 9.8l12.6 46.3c1.2 4.3 5.6 6.8 9.8 5.6l393.7-107.4C418.8 464.1 467.6 512 528 512c61.9 0 112-50.1 112-112V0H384zm144 448c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48z"
                />
              </svg>
              Order Tracking
            </button>
          </div>
          <span
            class="MuiTabs-indicator css-1qltlow-MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
      <div
        class="MuiBox-root css-eyevn2"
      >
        <form
          class="MuiBox-root css-yd8sa2"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-cmpglg-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r0:"
              id=":r0:-label"
            >
              Name
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ljffdk-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r0:"
                name="name"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-81qg8w"
                >
                  <span>
                    Name *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-ytlejw-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
            >
              Model
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-sc8y68-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              required=""
            >
              <div
                aria-controls=":r1:"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-model"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-w76bbz-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-component-select-model"
                role="combobox"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
                name="model"
                required=""
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-lohd6h-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-w4cd9x"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-cmpglg-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r2:"
              id=":r2:-label"
            >
              Colour
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ljffdk-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r2:"
                name="color"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-81qg8w"
                >
                  <span>
                    Colour *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-cmpglg-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for=":r3:"
              id=":r3:-label"
            >
              Quantity
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ljffdk-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r3:"
                name="quantity"
                required=""
                type="number"
                value="0"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-w1u3ce"
                >
                  <span>
                    Quantity *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiBox-root css-451ozx"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1jzxpmo-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Place Order
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      class="MuiBox-root css-1tvux47"
    >
      <div
        class="MuiBox-root css-xrblwu"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(76, 175, 80); font-size: 40px;"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
          />
        </svg>
        <div
          class="MuiBox-root css-0"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
          >
            Total Available Cars
          </h6>
          <h4
            class="MuiTypography-root MuiTypography-h4 css-1a9zl4x-MuiTypography-root"
          >
            0
          </h4>
        </div>
      </div>
      <div
        class="MuiBox-root css-xrblwu"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(255, 152, 0); font-size: 40px;"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
          />
        </svg>
        <div
          class="MuiBox-root css-0"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
          >
            Total Ordered Cars
          </h6>
          <h4
            class="MuiTypography-root MuiTypography-h4 css-1a9zl4x-MuiTypography-root"
          >
            0
          </h4>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
